pkg/volume/gce_pd/attacher.go

```diff
@@ -165,6 +165,9 @@ func (attacher *gcePersistentDiskAttacher) MountDevice(spec *volume.Spec, device
 	options := []string{}
 	if readOnly {
 		options = append(options, "ro")
+	} else {
+		// as per https://cloud.google.com/compute/docs/disks/add-persistent-disk#formatting
+		options = append(options, "discard")
```

Comment:
So you just made me look at the kernel source code again.

We should not run mkfs if we don't know that the device supports TRIM. 

You can see in fs/ext4/mballoc.c that if DISCARD is set but not supported in the device the kernel will emit a warning to dmesg, even just for mount.  I think if we cannot test for SSD or discard supported device we should not set this. 
There is no way to set mkfs flags per device?